src/ucp/core/ucp_am.c

```diff
@@ -698,7 +767,7 @@ ucp_am_send_req(ucp_request_t *req, size_t count,
          */
         max_short = -1;
     } else {
-        max_short = ucp_am_get_short_max(req, msg_config);
+        max_short = ucp_proto_get_short_max(req, max_short);
```

Comment:
wdyt about not changing `ucp_proto_get_short_max()` and use `msg_config`, but doing the following:
```
max_short = ucp_proto_get_short_max(req, msg_config);
if (max_short_limit != SSIZE_MAX) {
    max_short = ucs_min(max_short_limit, max_short);
}
```
I guess, compilers have to optimize this code, since `ucp_am_send_req()` declared with `inline` specification and we always pass a constant (i.e. `SSIZE_MAX` or `UCP_AM_SHORT_REPLY_MAX_SIZE`).
pass `SSIZE_MAX` to `ucp_am_send_req()` for non-AM_REPLY case

Reply:
in this case, if `ucp_proto_get_short_max` returns -1 (short is not supported due to some reason), `UCP_AM_SHORT_REPLY_MAX_SIZE` will be chosen for reply protocol